js/src/sheet.ts

```diff
@@ -396,7 +435,7 @@ let SheetView = widgets.DOMWidgetView.extend({
         this.model.save_changes();
     },
     _on_change: function(changes, source) {
-        if(this.hot === undefined || source == 'loadData') {
+        if(this.hot === undefined || source == 'loadData' || source == 'ObserveChanges.change') {
```

Comment:
This change is needed in order to prevent an infinite loop when having widgets in cells. I'm not sure why having widgets in cells causes an infinite loop but not having other cell types.